vm/safepoints.cpp

```diff
@@ -39,6 +39,15 @@ void safepoint_state::handle_safepoint(factor_vm* parent, cell pc) volatile {
     if (atomic::load(&parent->sampling_profiler_p))
       parent->end_sampling_profiler();
     std::cout << "Interrupted\n";
+    if (atomic::load(&parent->skip_debugger_p)) {
+      // Ctrl-Break throws an exception, interrupting the main thread, same
+      // as the "t" command in the factorbug debugger. But for Ctrl-Break to
+      // work we don't require the debugger to be activated, or even enabled.
+      atomic::store(&fep_p, false);
+      atomic::store(&parent->skip_debugger_p, false);
+      parent->general_error(ERROR_INTERRUPT, false_object, false_object);
+      FACTOR_ASSERT(false);
+    }
     parent->factorbug();
     atomic::store(&fep_p, false);
   } else if (atomic::load(&parent->sampling_profiler_p)) {
```

Comment:
Yes, and I know exactly why that is. It's in this code:
```
parent->factorbug();
atomic::store(&fep_p, false);
```
When you throw the "t", `factorbug()` throws an exception, the second line is not executed and `fep_p` is never set to `false`. In the Ctrl-Break thread I'm explicitly checking  for `fep_p` to avoid double interruption.
To work around that problem in my code I set `fep_p` to `false` before calling `parent->general_error`.

Since `factorbug()` can throw exceptions, the correct implementation would be to clear `fep_p` in a `finally` clause. Unfortunately, there [is no such thing](https://stackoverflow.com/questions/161177/does-c-support-finally-blocks-and-whats-this-raii-i-keep-hearing-about) in C++. Alternatively, we could at least clear `fep_p` before calling `parent->general_error` in `factorbug()`.